### PR TITLE
[DONTMERGE - WIP] First attempt to create DocBook 5 MAIN

### PR DESCRIPTION
**Pull request by tomschr**
* Add `chunker.xsl` from DocBook XSL
* Create `map-to-db5MAIN.xsl` to create DocBook5 output for MAIN file
* Implement previous stylesheet in `ditatodocbook.sh` script

### CODE REVIEW

File: map-to-db5MAIN.xsl [hunk 1]
```diff
@@ -0,0 +1,222 @@
+<?xml version="1.0"?>
```
Comment (ghost) [inline map-to-db5MAIN.xsl]: Please just change the existing map-to-MAIN to work with DB5 instead. I don't see much use in converting to DocBook 4 when we can have 5. (I.e. I think Dietrich if can create DB5, there is no DB4 support needed.)

File: ditatodocbook.sh [hunk 2]
```diff
@@ -138,13 +144,20 @@ mkdir -p "$outputxmldir"
...
-  --stringparam "remove" " $remove " \
-  "$mydir/map-to-MAIN.xsl" \
-  "$inputmap" > "$mainfile" 2> "$tmpdir/includes"
+  --stringparam includes.filename "$tmpdir/includes" \
+   ${MAP2MAIN_XSLT} \
+  "$inputmap"
+  # 2> "$tmpdir/includes"
+
+# Test, if the includes file is there, otherwise abort
+[[ -e "$tmpdir/includes" ]] || die "ERROR: File $tmpdir/includes not found"
```
Comment (ghost) [inline ditatodocbook.sh]: Use `-f` here -- I think `-e` would allow directories too

File: ditatodocbook.sh [hunk 3]
```diff
@@ -85,6 +90,7 @@ CLEANTEMP=1
 CLEANID=0
 TWEAK=""
 ENTITYFILE="entities.xml"
+MAP2MAIN_XSLT="$mydir/map-to-MAIN.xsl"
```
Comment (ghost) [inline ditatodocbook.sh]: What is this good for? Especially among the "configurable" options, this seems misplaced.

**Review by ghost — CHANGES_REQUESTED** [review]
Just as a preliminary review for this.

**Comment by ghost** [comment]
I've had a look at the new `map-to-db5MAIN` file. 

* I diffed between `map-to-MAIN` and `map-to-db5MAIN` (see 568fa54085f26924cfc37f76dcd453662ac4009a). Getting a 600-line diff for a 250-line file is ... unexpected. Most of the diffed lines are (unwanted) whitespace changes.
* Your reworking currently partially destroys the features to remove/replace files at the top level. To make this work properly, you would need to copy/paste the relevant parts of normal-mode templates into the collect.files templates.
* It is also incomplete, there is still an xsl:message for content that needs to go into `includes` at the end left.

This is all stuff that had already been working.

If this is just about seeing error and warning messages on the command line, it seems to me that there are easier ways to get those (like just outputting the messages caught in includes after the xsltproc run).
That may not be proper-est way to go about this but it would be quick.

File: ditatodocbook.sh (hunk 3 above)
Comment (tomschr) [reply ditatodocbook.sh]: It's just my style. 😉  I don't want to repeat nor memorize the name of the stylesheet.

File: ditatodocbook.sh (hunk 2 above)
Comment (tomschr) [reply ditatodocbook.sh]: Yes, good idea! 👍

File: ditatodocbook.sh (hunk 3 above)
Comment (ghost) [reply ditatodocbook.sh]: This is only used once. And it definitely should not be grouped together with the user-configurable options.

File: docbook/topic2db.xsl [hunk 4]
```diff
@@ -115,13 +122,13 @@
...
     <xsl:variable name="elementName">
       <xsl:choose>
-      <xsl:when test="$contextType='article'">artheader</xsl:when>
-      <xsl:when test="$contextType='appendix'">docinfo</xsl:when>
-      <xsl:when test="$contextType='book'">bookinfo</xsl:when>
-      <xsl:when test="$contextType='chapter'">chapterinfo</xsl:when>
-      <xsl:when test="$contextType='glossary'">docinfo</xsl:when>
-      <xsl:when test="$contextType='part'">docinfo</xsl:when>
-      <xsl:when test="$contextType='section'">sectioninfo</xsl:when>
+      <xsl:when test="$contextType='article'">info</xsl:when>
```
Comment (ghost) [inline docbook/topic2db.xsl]: I think this when is unnecessary now.

**Comment by ghost** [comment]
Hm, there still seem to be a few issues here that I didn't figure out:

* we're somehow writing the DB5 namespaces in lots of elements, not just the root element
* daps xmlformat does not run, the error is `ERROR: Failed to get filelist for ROOTID ""`
* there is still an namespace-free xpath looking for `@linkend`

Also, I have moved your map/bookmap commit onto https://github.com/sknorr/dietrich/tree/f/map-to-bookmap

**Pull request merged**